src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -3082,7 +3082,18 @@ private void UpdateDatabasesWithNewServerWideBackupConfiguration(TransactionOper
                     {
                         foreach (BlittableJsonReaderObject backup in backups)
                         {
-                            if (IsServerWideBackupWithTaskName(backup, periodicBackupConfiguration.Name))
+                            if (backup.TryGet(nameof(PeriodicBackupConfiguration.Name), out string backupName)  == false || backupName == null)
+                                throw new RachisInvalidOperationException(
+                                    $"This {nameof(PeriodicBackupConfiguration)} of {result.Key} has no value for {nameof(PeriodicBackupConfiguration.Name)}. This should not happens\n{backup}");
+                            
+                            if (serverWideBackupConfiguration.ToRemoveFromDatabaseRecord != null)
+                            {
+                                var toRemove = PutServerWideBackupConfigurationCommand.GetTaskNameForDatabase(serverWideBackupConfiguration.ToRemoveFromDatabaseRecord);
+                                if(backupName.Equals(toRemove, StringComparison.OrdinalIgnoreCase))
+                                    continue;
+                            }
+                                
+                            if(backupName.Equals(periodicBackupConfiguration.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
Shouldn't we rely on `TaskId` instead of `Name` to detect that this is an update?

Reply:
The `TaskId` from the server-wide backup can be different from the `TaskId` from the database records
https://github.com/ravendb/ravendb/pull/11086#discussion_r484883827